lib/ansible/modules/cloud/vmware/vmware_guest_facts.py

```diff
@@ -130,15 +124,23 @@ def main():
         name=dict(type='str'),
         name_match=dict(type='str', choices=['first', 'last'], default='first'),
         uuid=dict(type='str'),
-        folder=dict(type='str', default='/vm'),
+        folder=dict(type='str'),
         datacenter=dict(type='str', required=True),
     )
     module = AnsibleModule(argument_spec=argument_spec,
                            required_one_of=[['name', 'uuid']])
 
-    # FindByInventoryPath() does not require an absolute path
-    # so we should leave the input folder path unmodified
-    module.params['folder'] = module.params['folder'].rstrip('/')
+    # build the search path. If a "folder" is given AND is an
+    # absolute path, pretend the user knows what he/she does
+    # and use that path. Else, prepend the "folder" path
+    # with "/<name_of_datacenter>/vm"
+    if module.params['folder']:
+        if module.params['folder'].startswith('/'):
+            pass
+        else:
+            module.params['folder'] = '/' + module.params['datacenter'] + '/vm/' + module.params['folder']
```

Comment:
What if user provides `vm` as folder path, then path is created as `/DC0/vm/vm/` and Findinventorybypath will fail.

Reply:
True, but therefore they should either provide a relativ path to the 'root' folder (which is '/<datacenter>/vm' by default) or an absolute name. Then they should know, what they are doing.

At first, I did only change the code to only use "relativ" paths, because we already have to provide "datacenter". But then, the integration tests failed, because it uses an absolute path for its tests.

Currently, it always fails when your virtual machines are stored in the "default" root folder and you do not provide a "folder" at all. Then, it searches within "/vm" which is invalid. See the first comment and watch the "folder" string.